static-analysis/action.yaml

```diff
@@ -13,16 +9,12 @@ runs:
   steps:
     - uses: actions/checkout@v4
       with:
-        fetch-depth: 2 # Fetch only the last two commits for efficient diff comparison
+        fetch-depth: 2
 
     - run: |
-        docker run --rm -v "${PWD}:/src" \
-        -e SEMGREP_APP_TOKEN=${{ inputs.semgrep-app-token }} \
-        -e SEMGREP_REPO_NAME=${GITHUB_REPOSITORY} \
-        -e SEMGREP_BRANCH=${GITHUB_REF} \
-        -e SEMGREP_COMMIT=${{ github.event.pull_request.head.sha }} \
-        -e SEMGREP_PR_ID=${{ github.event.pull_request.number }} \
-        -e SEMGREP_BASELINE_REF='HEAD^' \
-        semgrep/semgrep:latest-nonroot \
-        semgrep ci
+        export PATH=$PATH:$HOME/.local/bin
+
+        pipx install --force semgrep  # Install Semgrep if not already available
```

Comment:
You will need to install python before this runs using `action-setup-tools` and specify a specific python version

Reply:
Added the action setup tools to install the python before installing semgrep.